cdc/metadata.cc

```diff
@@ -65,6 +65,13 @@ cdc::metadata::container_t::const_iterator cdc::metadata::gen_used_at(api::times
     return std::prev(it);
```

Comment:
re commit message:

> We can do it without losing data consistency because clients read CDC log tables with at least 10 s delay.

We can write as far into the past as we like without breaking any data consistency, if we know those past generations (we haven't removed them).

There is only a problem with future generations, because we don't know what they'll be. Hence `generation_leeway`, representing the assumption that we'll learn about a new generation with timestamp `ts` before we reach `ts - generation_leeway`.

The thing with CDC clients is independent of generations.
After all, if the generation didn't change for 30 minutes, we do allow a write with timestamp = `now - 29 minutes`.

Reply:
I removed this sentence. I did the same in the cover letter and `cdc.md`.